pystac_client/mixins.py

```diff
@@ -67,3 +67,14 @@ def _get_queryables_href(self) -> str:
         link = self.get_single_link(QUERYABLES_REL)
         href = self._get_href(QUERYABLES_REL, link, QUERYABLES_ENDPOINT)
         return href
+
+    def write_queryables_to(self, queryables: Dict[str, Any], url: str) -> None:
+        """Write a queryables to a given endpoint"""
+        if self._stac_io is None:
+            raise APIError("API access is not properly configured")
+        self._stac_io.save_json(url, queryables)
+
+    def set_queryables(self, queryables: Dict[str, Any]) -> None:
+        """Write a queryables to the default endpoint"""
+        url = self._get_queryables_href()
+        self.write_queryables_to(queryables, url)
```

Comment:
I don't think we need to add two methods to the public API surface, my recommendation is to consolidate to one:

```suggestion
    def set_queryables(self, queryables: Dict[str, Any], url: Option[str]) -> None:
        """Write a queryables to the an endpoint.
        
        Defaults to the queryables href as defined by the STAC API extension.
        """
        if self._stac_io is None:
            raise APIError("API access is not properly configured (`_stac_io` not set)")
        if url is None:
            url = self._get_queryables_href()
        self._stac_io.save_json(url, queryables)
```

Reply:
>  how hard would it be to add a general transaction-enabled StacIO subclass to this PR
This should be straight forward, but I think this is a PR of its own.

> Is writing to queryables documented anywhere?
No, I also could not find any references. However, there needs to be away to update queryables, in particular since collection queryables may differ from catalog queryables. What is the best approach here? Make a PR to the filter extension and propose an optional `POST`/ `PUT` method for the queryables endpoint?

 > my recommendation is to consolidate to one
 Sure, can do. I was mostly mirroring the `get_queryable` method

Reply:
i was mostly mirroring the approach of the `get_queryables` method